### PR TITLE
DDF-2698 Remove -d alias from Export Command

### PR DESCRIPTION
**Pull request by rzwiefel**
#### What does this PR do?
 - Removes `-d` delete alias from export command
 - Short circuits trying to sign zip if nothing was exported
 - Added unit tests for Export Command
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
@adimka @bdeining 

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).

@coyotesqrl
@lessarderic

#### How should this be tested? (List steps with links to updated documentation)
1. Ensure -d command has no conflicts
2. Ensure on an export with no metacards or content there are no errors in the log
#### What are the relevant tickets?
https://codice.atlassian.net/browse/DDF-2698

#### Checklist:
- [x] Update / Add Unit Tests

### CODE REVIEW

**Comment by jlcsmith** [comment]
Please address checkstyle failure in catalog-core-commands

File: catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,283 @@
+package org.codice.ddf.commands.catalog
```
Comment (jlcsmith) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: Missing header

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java [hunk 2]
```diff
@@ -143,20 +144,20 @@ protected Object executeWithSubject() throws Exception {
         final File outputFile = initOutputFile(output);
         if (outputFile.exists()) {
             printErrorMessage(String.format("File [%s] already exists!", outputFile.getPath()));
-            return null;
```
Comment (coyotesqrl) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java]: Why this change? In all our other commands, we simply return after printing the error message. What advantage are you gaining from the runtime exception?

Is this change here just to support unit tests?

What is the effect on the runtime system?

File: catalog/core/catalog-core-commands/pom.xml [hunk 3]
```diff
@@ -134,6 +134,12 @@
             <artifactId>catalog-transformer-zip</artifactId>
             <version>${project.version}</version>
         </dependency>
+        <dependency>
```
Comment (coyotesqrl) [inline catalog/core/catalog-core-commands/pom.xml]: Did coverage go up or down? It seems hard to believe it held steady with a new class and new test class for it.

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java (hunk 2 above)
Comment (rzwiefel) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java]: Absolutely. I'd rather use exceptions as a means of exceptional state than trying to parse log messages? Unless you have any other ideas.

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java [hunk 4]
```diff
@@ -143,20 +144,20 @@ protected Object executeWithSubject() throws Exception {
...
+            throw new IllegalStateException("File already exists!");
         }
 
         final File parentDirectory = outputFile.getParentFile();
         if (parentDirectory == null || !parentDirectory.isDirectory()) {
             printErrorMessage(String.format("Directory [%s] must exist.", output));
             console.println("If the directory does indeed exist, try putting the path in quotes.");
-            return null;
+            throw new IllegalStateException("Must be inside of a directory!");
```
Comment (bdeining) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java]: nit : We generally haven't been using `!` in messages

File: catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+    }
+
+    Map namespaces = ['xmlns'         : 'urn:catalog:metacard',
+                      'xmlns:gml'     : 'http://www.opengis.net/gml',
+                      'xmlns:xlink'   : 'http://www.w3.org/1999/xlink',
+                      'xmlns:smil'    : 'http://www.w3.org/2001/SMIL20/',
+                      'xmlns:smillang': 'http://www.w3.org/2001/SMIL20/Language']
+
+    Closure metacardToXmlTransformer = {
+            //    BiFunction<Metacard, Map, BinaryContent> metacardToXmlTransformer = {
```
Comment (bdeining) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: Remove comment?

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java [hunk 6]
```diff
@@ -115,8 +117,7 @@
     String output = Paths.get(System.getProperty("ddf.home"), FILE_NAMER.get())
             .toString();
 
-    @Option(name = "--delete", required = true, aliases = {
-            "-d"}, multiValued = false, description = "Delete Metacards and content after export. EG: --delete=true or --delete=false")
+    @Option(name = "--delete", required = true, multiValued = false, description = "Delete Metacards and content after export. EG: --delete=true or --delete=false")
```
Comment (bdeining) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java]: Instead of having to specify `--delete=true` or `--delete=false`,  is there a way to refactor this to simply be `--delete` in the case where we want to delete Metacards and content after export, otherwise don't delete when no flag is specified?

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java (hunk 2 above)
Comment (coyotesqrl) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java]: I don't have other ideas for integrating with your unit tests, but I'm more concerned right now about the runtime impact.

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java (hunk 6 above)
Comment (rzwiefel) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java]: Purposely made it required so the user has to put it every time. You still can use `--delete` as the short implicit form of `--delete=true`.

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java (hunk 2 above)
Comment (rzwiefel) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java]: No runtime impact other than it exits the command errorfully instead of with zero information about what went wrong.

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java [hunk 7]
```diff
@@ -545,4 +552,11 @@ private QueryRequestImpl getQuery(Filter filter, int index, int pageSize) {
                 TimeUnit.MINUTES.toMillis(1)), new HashMap<>());
     }
 
+    public void setStorageProvider(StorageProvider storageProvider) {
```
Comment (lessarderic) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java]: ❓ Do we need `setStorageProvider` and `setJarSigner`? Doesn't look like they are used anywhere.

File: catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy [hunk 8]
```diff
@@ -0,0 +1,283 @@
...
+            throw new ResourceNotFoundException('Could not find exception')
+        }
+
+        exportCommand = new ExportCommand(filterBuilder: new GeotoolsFilterBuilder(),
+                bundleContext: bundleContext, catalogFramework: catalogFramework)
+
+    }
+
+    void cleanup() {
+        assert tmpHomeDir?.deleteDir()
```
Comment (lessarderic) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: ❓ Why the `assert` here?

File: catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy [hunk 9]
```diff
@@ -0,0 +1,283 @@
...
+        setup:
+        exportCommand.with {
+            delete = false
+        }
+
+        when:
+        exportCommand.executeWithSubject()
+
+        then:
+        notThrown(Exception)
```
Comment (lessarderic) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: ❓ Should we also verify that no zip file was created?

File: catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy [hunk 10]
```diff
@@ -0,0 +1,283 @@
...
+        println(files)
+        assert [result.metacard.id].every { id ->
+            files.any { it.contains(id) }
+        }
+        assert [resourceName].every { name ->
+            files.any { it.contains(name) }
+        }
+
+    }
+
```
Comment (lessarderic) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: ❗️ We should add the following tests:
* Xml transformer missing (L137)
* Parent directory is `null`/doesn't exist (L151)
* File name is blank (L158)
* Metacards and content are deleted when the `--delete` flag is used

File: catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy [hunk 11]
```diff
@@ -0,0 +1,283 @@
...
+        exportCommand.with {
+            delete = false
+            output = file.canonicalPath
+        }
+
+        when:
+        exportCommand.executeWithSubject()
+
+        then:
+        thrown(IllegalStateException)
```
Comment (lessarderic) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: ❓ Should we make sure that the existing file wasn't touched?

File: catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy [hunk 12]
```diff
@@ -0,0 +1,283 @@
...
+        exportCommand.with {
+            it.delete = true
+            it.session = session
+        }
+
+        when:
+        exportCommand.executeWithSubject()
+
+        then:
+        notThrown(Exception)
```
Comment (lessarderic) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: ❓ Do we really need to assert that no exception was thrown when we exception success? I would imagine that if one is thrown and we don't have the assertion, the test would still fail?

File: catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy [hunk 13]
```diff
@@ -0,0 +1,283 @@
...
+        }
+
+        when:
+        exportCommand.executeWithSubject()
+
+        then:
+        thrown(IllegalStateException)
+    }
+
+    def "Test abort command"() {
```
Comment (lessarderic) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: ❓ Do we really need to test this? Feels like we're testing the Karaf console more than the actual command. Thoughts?

File: catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy [hunk 14]
```diff
@@ -0,0 +1,283 @@
...
+        tmpHomeDir.list().find()?.endsWith('.zip')
+
+        String zip = tmpHomeDir.listFiles().find()?.canonicalPath
+        assert zip?.trim() as boolean // null or empty
+        assert zip.endsWith('.zip')
+
+        def files = new ZipFile(zip).entries()
+                .collect { it.isDirectory() ? null : it.name }
+                .findAll { it != null }
+        println(files)
```
Comment (lessarderic) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: ❗️ Remove

File: catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy [hunk 15]
```diff
@@ -0,0 +1,283 @@
...
+
+    }
+
+/**************************************************************************
+ *
+ * Utility Methods
+ *
+ *************************************************************************/
+
+    Metacard simpleMetacard(Map kwargs) {
```
Comment (lessarderic) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: ✏️  `attributes` instead of `kwargs`?

File: catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy [hunk 16]
```diff
@@ -0,0 +1,283 @@
...
+        String id = randomUUID()
+        [
+                (Metacard.ID)          : id,
+                (Metacard.TITLE)       : 'Metacard Title',
+                (Metacard.RESOURCE_URI): "content:$id".toString(),
+        ]
+    }
+
+    String randomUUID() {
+        UUID.randomUUID().toString().replaceAll('-', '')
```
Comment (lessarderic) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: ❓ Do we really need to do this since we're mocking out the Catalog?

File: catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy [hunk 17]
```diff
@@ -0,0 +1,283 @@
...
+    void setup() {
+        tmpHomeDir = File.createTempDir()
+        System.setProperty("ddf.home", tmpHomeDir.canonicalPath)
+
+        ServiceReference xmlTransformerReference = Mock(ServiceReference)
+
+        xmlTransformer = Mock(MetacardTransformer)
+
+        xmlTransformer.transform(_ as Metacard, _ as Map) >> { metacard, map ->
+            metacardToXmlTransformer(metacard, map)
```
Comment (lessarderic) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: ❓ Instead of having a closure that does a "real" transformation, can't we just have the `transform` method return canned responses? That would eliminate the chance of having to deal with bugs in the `metacardToXmlTransformer` closure.

**Comment by lessarderic** [comment]
@rzwiefel, any update on this PR?

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java [hunk 18]
```diff
@@ -143,20 +144,20 @@ protected Object executeWithSubject() throws Exception {
...
             console.println("If the directory does indeed exist, try putting the path in quotes.");
-            return null;
+            throw new IllegalStateException("Must be inside of a directory!");
         }
 
         String filename = FilenameUtils.getName(outputFile.getPath());
         if (StringUtils.isBlank(filename) || !filename.endsWith(".zip")) {
             console.println("Filename must end with '.zip' and not be blank");
-            return null;
+            throw new IllegalStateException("Must not be blank and end with '.zip'!");
```
Comment (ricklarsen) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java]: Message would be clearer as `Must not be blank and must end with '.zip'`

File: catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy (hunk 16 above)
Comment (rzwiefel) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: I just need a random ID. I prefer random over a hardcoded so that way you ensure just in case there's some weird state carryover between tests that you are using a new random ID so there will never be conflict.

File: catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy (hunk 10 above)
Comment (rzwiefel) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: - added missing xformer test
- Cannot produce null parent directory
- added blank filename test
- added test to ensure deletions are called

File: catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy (hunk 12 above)
Comment (rzwiefel) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: Correct the test would still fail. It is best practice (and IMO looks better + is more explicit) to at a minimum list the single expectation for the test -- That nothing went wrong.

File: catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy (hunk 13 above)
Comment (rzwiefel) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: Since it is up to each command to prompt the user and read the input I would say it is important. If when refactoring this file someone moves the if statement containing the force prompt, they might inadvertently remove the force check which this test would catch.

File: catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy (hunk 11 above)
Comment (rzwiefel) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: done

File: catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy (hunk 9 above)
Comment (rzwiefel) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: done

File: catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy (hunk 8 above)
Comment (rzwiefel) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: `deleteDir` returns true if the delete succeeds (or the folder doesn't exist) and false if it fails (or isn't a directory). This way ensure each test is cleaned up and have no false negative failures because state that wasn't cleaned up... Ironically it was unable to delete the directory every time causing failures... removed the assert so we best effort delete (but still temp file so should get cleaned up)

File: catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy (hunk 17 above)
Comment (rzwiefel) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: removed

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java (hunk 7 above)
Comment (rzwiefel) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java]: Didn't know if they were still needed when using the `@Reference` annotation for injection. Will remove

File: catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy (hunk 5 above)
Comment (rzwiefel) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: done

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java (hunk 4 above)
Comment (rzwiefel) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java]: done

File: catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy (hunk 15 above)
Comment (rzwiefel) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: done

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java (hunk 18 above)
Comment (rzwiefel) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java]: done

**Comment by rzwiefel** [comment]
build now

**Comment by rzwiefel** [comment]
Ready for review

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java [hunk 19]
```diff
@@ -178,8 +179,13 @@ protected Object executeWithSubject() throws Exception {
         console.println("Starting metacard export...");
         Instant start = Instant.now();
         List<ExportItem> exportedItems = doMetacardExport(zipFile, filter);
-        console.println("Metacards exported in: " + getFormattedDuration(start));
+        if (exportedItems.isEmpty()) {
+            console.println("No metacards found to export, exiting.");
+            FileUtils.deleteQuietly(zipFile.getFile());
```
Comment (coyotesqrl) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java]: ❓ I believe this is the only place where you close the file (implicitly, with the `deleteQuietly`), so in the case where there are metacards exported, you'll leave it open. Might be best to use try-with-resources or wrap in a try/finally.

Comment (rzwiefel) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java]: As discussed, all opening and closing is handled within the ZipFile itself and not accessible externally. (ZipFile is not related to or descended from File in any way)

File: catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy (hunk 1 above)
Comment (bdeining) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: Still need a header?

File: catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy [hunk 20]
```diff
@@ -0,0 +1,351 @@
...
+
+        def files = new ZipFile(zip).entries()
+                .collect { it.isDirectory() ? null : it.name }
+                .findAll { it != null }
+
+        assert [result.metacard.id].every { id ->
+            files.any { it.contains(id) }
+        }
+
+
```
Comment (bdeining) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: extra whitespace

File: catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy [hunk 21]
```diff
@@ -0,0 +1,351 @@
...
+    def "Test export no transformer"() {
+        setup:
+        def bundleContext = Mock(BundleContext) {
+            getServiceReferences(_, _) >> []
+        }
+        def exportCommand = new ExportCommand(filterBuilder: new GeotoolsFilterBuilder(),
+                bundleContext: bundleContext, catalogFramework: catalogFramework)
+
+
```
Comment (bdeining) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: nit : extra whitespace

Comment (rzwiefel) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: fixed

File: catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy (hunk 20 above)
Comment (rzwiefel) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: fixed

**Comment by rzwiefel** [comment]
build now

File: catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy (hunk 1 above)
Comment (coyotesqrl) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: ❗️ Yep. Still needs the header.

Comment (rzwiefel) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: fixed

**Comment by rzwiefel** [comment]
build now

**Comment by rzwiefel** [comment]
@jlcsmith @lessarderic @bdeining @adimka Ready for re-review

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java [hunk 22]
```diff
@@ -420,8 +427,23 @@ private void writeToZip(/*Mutable,IO*/ ZipFile zipFile, ExportItem exportItem,
...
-        zipFile.addStream(resource.getResource()
-                .getInputStream(), parameters);
+        try (InputStream resourceStream = resource.getResource()
+                .getInputStream()) {
+            zipFile.addStream(resourceStream, parameters);
+        } catch (IOException e) {
+            LOGGER.warn("Could not get content. content will not be added to zip [{}]",
+                    exportItem.getId());
+            console.printf(
+                    "%sCould not transform metacard. Metacard will not be included in export. %s (%s)%s%n",
```
Comment (jlcsmith) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java]: ❓ shouldn't this message match the one above?

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java [hunk 23]
```diff
@@ -145,20 +146,20 @@ protected Object executeWithSubject() throws Exception {
...
             console.println("If the directory does indeed exist, try putting the path in quotes.");
-            return null;
+            throw new IllegalStateException("Must be inside of a directory");
         }
 
         String filename = FilenameUtils.getName(outputFile.getPath());
         if (StringUtils.isBlank(filename) || !filename.endsWith(".zip")) {
             console.println("Filename must end with '.zip' and not be blank");
-            return null;
+            throw new IllegalStateException("Must not be blank and must end with '.zip'");
```
Comment (jlcsmith) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java]: ❓ should this include "Filename"?  Eg, "**Filename** must not be ..."

Comment (rzwiefel) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java]: done

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java (hunk 22 above)
Comment (rzwiefel) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java]: done

**Comment by rzwiefel** [comment]
build now

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java [hunk 24]
```diff
@@ -110,15 +112,14 @@
...
+    protected StorageProvider storageProvider;
 
     @Option(name = "--output", description = "Output file to export Metacards and contents into. Paths are absolute and must be in quotes. Will default to auto generated name inside of ddf.home", multiValued = false, required = false, aliases = {
             "-o"})
     String output = Paths.get(System.getProperty("ddf.home"), FILE_NAMER.get())
             .toString();
 
-    @Option(name = "--delete", required = true, aliases = {
-            "-d"}, multiValued = false, description = "Delete Metacards and content after export. EG: --delete=true or --delete=false")
+    @Option(name = "--delete", required = true, multiValued = false, description = "Delete Metacards and content after export. E.g., --delete=true or --delete=false")
```
Comment (rzwiefel) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java]: Because I designed the command that way, thus there is no ambiguity ever whether or not things are being deleted or whether the capability exists to delete them.

Comment (rzwiefel) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java]: It's like quickbuilding... using an alias means its harder when your not on something with your setup (and won't remember what to put), its more confusing to new people who dont realize what aliased commands are doing, etc.

I absolutely prefer making every person to run this command type this out **every single time**.

File: catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy (hunk 14 above)
Comment (rzwiefel) [inline catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy]: done
